dags/hindawi/parser.py

```diff
@@ -0,0 +1,217 @@
+            ),
+            CustomExtractor(
+                destination="authors",
+                extraction_function=self._get_authors,
+                default_value=[],
+            ),
+            TextExtractor(
+                destination="abstract",
+                source="ns0:metadata/ns1:record/ns0:datafield/[@tag='520']/ns0:subfield/[@code='a']",
+                prefixes=self.prefixes,
+                extra_function=lambda x: " ".join(x.split()),
+                default_value="",
+            ),
+            TextExtractor(
+                destination="title",
+                source="ns0:metadata/ns1:record/ns0:datafield/[@tag='245']/ns0:subfield/[@code='a']",
+                prefixes=self.prefixes,
+                extra_function=lambda x: x,
+                default_value="",
+            ),
+            TextExtractor(
+                destination="date_published",
+                source="ns0:metadata/ns1:record/ns0:datafield/[@tag='260']/ns0:subfield/[@code='c']",
+                prefixes=self.prefixes,
+                extra_function=lambda x: x,
+                default_value="2000-00-00",
+            ),
+            TextExtractor(
+                destination="page_nr",
+                source="ns0:metadata/ns1:record/ns0:datafield/[@tag='300']/ns0:subfield/[@code='a']",
+                prefixes=self.prefixes,
+                extra_function=lambda x: [int(x)],
+                default_value=[0],
+            ),
+            CustomExtractor(
+                destination="publication_info",
+                extraction_function=self._get_publication_info,
+                default_value=[],
+            ),
+            CustomExtractor(
+                destination="arxiv_eprints",
+                extraction_function=self._get_arxiv,
+                default_value=[],
+            ),
+            CustomExtractor(
+                destination="copyright_statement",
+                extraction_function=self._get_copyright_statement,
+                default_value="",
+            ),
+            CustomExtractor(
+                destination="copyright_year",
+                extraction_function=self._get_copyright_year,
+                default_value=0,
+            ),
+            CustomExtractor(
+                destination="license",
+                extraction_function=self._get_license,
+                default_value=[],
+            ),
+        ]
+        super().__init__(extractors)
+
+    def _get_authors(self, article: ET.Element):
+        authors_all = article.findall(
+            "ns0:metadata/ns1:record/ns0:datafield/[@tag='100']", self.prefixes
+        ) + article.findall(
+            "ns0:metadata/ns1:record/ns0:datafield/[@tag='700']", self.prefixes
+        )
+
+        authors = []
+        for author in authors_all:
+            raw_name = author.find("ns0:subfield[@code='a']", self.prefixes).text
+            if not raw_name:
+                continue
+            author_ = {
```

Comment:
maybe a better name?

Reply:
This fix has to go to another branch, because this PR is for IOP, not hindawi publisher